src/main/kotlin/org/danilopianini/plagiarismdetector/analyzer/Analyzer.kt

```diff
@@ -0,0 +1,18 @@
+package org.danilopianini.plagiarismdetector.analyzer
+
+import org.danilopianini.plagiarismdetector.analyzer.representation.SourceRepresentation
+
+/**
+ * An interface modeling the analyzer, the component that analyze the source
+ * code producing in output a comparable representation of the source file.
+ * @param <I> the type of the source file given in input
+ * @param <O> the type of output
+ * @param <T> the type of the representation
+ */
+interface Analyzer<I, out O : SourceRepresentation<T>, T> {
```

Comment:
This might maybe just extend a normal function:

interface Analyzer<in I, out O : SourceRepresentation<T>, T> : (I) -> O